src/options/Makefile.am

```diff
@@ -262,179 +108,44 @@ nodist_liboptions_la_SOURCES = \
-#	echo heads
-#	echo "$@"
-#	echo $(OPTIONS_HEADS)
-	$(AM_V_at)chmod +x @srcdir@/mkoptions
-	$(AM_V_GEN)(@srcdir@/mkoptions apply-sed-files-to-template \
-		$(if $(COVERAGE_ON), @abs_srcdir@, @srcdir@)/base_options_template.h \
-		base_options_template.h.sed \
-		"$<" \
-	) > "$@"
-
-summary.sed : mkoptions $(OPTIONS_OPTIONS_FILES)
-	$(AM_V_at)chmod +x @srcdir@/mkoptions
-	$(AM_V_GEN)(@srcdir@/mkoptions summary-sed \
-		$(OPTIONS_OPTIONS_FILES) \
-	) > summary.sed
-
-
-
-#   mkoptions apply-sed-to-template sed-file template-file
-options_holder.h : options_holder_template.h options_holder_template.h.sed summary.sed mkoptions $(OPTIONS_HEADS)
-	$(AM_V_at)chmod +x @srcdir@/mkoptions
-	$(AM_V_GEN)(@srcdir@/mkoptions apply-sed-files-to-template \
-		$(if $(COVERAGE_ON), @abs_srcdir@, @srcdir@)/options_holder_template.h \
-		@builddir@/options_holder_template.h.sed \
-		summary.sed \
-	) > "$@"
-
-# Make sure not to match with "options.cpp" too.
-%_options.cpp: %_options.sed %_options.h mkoptions options_holder.h base_options_template.cpp base_options_template.cpp.sed
-	$(AM_V_at)chmod +x @srcdir@/mkoptions
-	$(AM_V_GEN)(@srcdir@/mkoptions apply-sed-files-to-template \
-		$(if $(COVERAGE_ON), @abs_srcdir@, @srcdir@)/base_options_template.cpp \
-		base_options_template.cpp.sed \
-		"$<" \
-	) > "$@"
-
-
-
-
-#   mkoptions apply-sed-to-template sed-file template-file
-options.cpp : options_template.cpp options_template.cpp.sed mkoptions summary.sed $(OPTIONS_HEADS) options_holder.h
-	$(AM_V_at)chmod +x @srcdir@/mkoptions
-	$(AM_V_GEN)(@srcdir@/mkoptions apply-sed-files-to-template \
-		$(if $(COVERAGE_ON), @abs_srcdir@, @srcdir@)/options_template.cpp \
-		@builddir@/options_template.cpp.sed \
-		summary.sed \
-	) > "$@"
-
-
-#   mkoptions apply-sed-to-template sed-file template-file
-options_get_option.cpp : options_get_option_template.cpp options_get_option_template.cpp.sed mkoptions summary.sed $(OPTIONS_HEADS)
-	$(AM_V_at)chmod +x @srcdir@/mkoptions
-	$(AM_V_GEN)(@srcdir@/mkoptions apply-sed-files-to-template \
-		$(if $(COVERAGE_ON), @abs_srcdir@, @srcdir@)/options_get_option_template.cpp \
-		@builddir@/options_get_option_template.cpp.sed \
-		summary.sed \
-	) > "$@"
-
-options_set_option.cpp : options_set_option_template.cpp options_set_option_template.cpp.sed mkoptions summary.sed $(OPTIONS_HEADS)
-	$(AM_V_at)chmod +x @srcdir@/mkoptions
-	$(AM_V_GEN)(@srcdir@/mkoptions apply-sed-files-to-template \
-		$(if $(COVERAGE_ON), @abs_srcdir@, @srcdir@)/options_set_option_template.cpp \
-		@builddir@/options_set_option_template.cpp.sed \
-		summary.sed \
-	) > "$@"
-
-
-
-$(DOCUMENTATION_FILES) : % : %_template %_template.sed mkoptions summary.sed
-#	echo "$<"
-#	echo "$@"
-	$(AM_V_at)chmod +x @srcdir@/mkoptions
-	$(AM_V_GEN)(@srcdir@/mkoptions apply-sed-files-to-template \
-		"$<" \
-		"$<".sed \
-		@builddir@/summary.sed \
-	) > "$@"
-
-
+$(CPP_TEMPLATE_FILES):;
```

Comment:
What is this line doing?

Reply:
The comment above says: 
# Make sure the implicit rules never mistake a _template.cpp or _template.h file for source file.

This rule was already in the old version of the Makefile.am, looks hackish, but there was probably a reason why it was added.